Robust.Client/UserInterface/CustomControls/DebugConsole.cs

```diff
@@ -204,6 +212,83 @@ private void CommandBarOnOnKeyBindDown(GUIBoundKeyEventArgs args)
                 Output.ScrollToBottom();
                 args.Handle();
             }
+            else if (args.Function == EngineKeyFunctions.GuiTabNavigateNext)
+            {
+                NextCommand();
+                args.Handle();
+                return;
+            }
+            else if(args.Function == EngineKeyFunctions.GuiTabNavigatePrev)
+            {
+                PrevCommand();
+                args.Handle();
+                return;
+            }
+        }
+
+        private void SetInput(string cmd)
+        {
+            CommandBar.Text = cmd;
+            CommandBar.CursorPosition = cmd.Length;
+        }
+        
+        private void FindCommands()
+        {
+            searchResults.Clear();
+            searchIndex = 0;
+            commandChanged = false;
+            foreach (var cmd in Commands)
+            {
+                if (cmd.Key.StartsWith(CommandBar.Text))
+                {
+                    searchResults.Add(cmd.Key);
+                }
+            }
+        }
+
+        private void NextCommand()
+        {
+            if (!commandChanged)
+            {
+                if (searchResults.Count == 0)
+                    return;
+
+                searchIndex = ++searchIndex % searchResults.Count;
```

Comment:
Don't use `++` here since you're not making use of the actual mutate it does.
```suggestion
                searchIndex = (searchIndex + 1) % searchResults.Count;
```